Readme.md

```diff
@@ -2,31 +2,106 @@
+libbpfgo is built around [libbpf](https://github.com/libbpf/libbpf) - the
+standard library for interacting with eBPF programs from userspace - which is a
+C library maintained in Linux upstream. We have created libbpfgo as a thin Go
+wrapper around the libbpf project.
 
 ## Installing
 
-libbpfgo is using CGO to interop with libbpf and will expect to be linked with libbpf at run or link time. Simply importing libbpfgo is not enough to get started, and you will need to fulfill the required dependency in one of the following ways:
+libbpfgo uses CGO to interop with libbpf and will expect to be linked with
+libbpf at run or link time. Simply importing libbpfgo is not enough to get
+started, and you will need to fulfill the required dependency in one of the
+following ways:
 
-1. Install the libbpf as a shared object in the system. Libbpf may already be packaged for you distribution, if not, you can build and install from source. More info [here](https://github.com/libbpf/libbpf).
-1. Embed libbpf into your Go project as a vendored dependency. This means that the libbpf code is statically linked into the resulting binary, and there are no runtime dependencies. [Tracee](https://github.com/aquasecurity/tracee) takes this approach and you can take example from it's [Makefile](https://github.com/aquasecurity/tracee/blob/f8df7da6a27f729610992b6bd52e89d510fcf384/tracee-ebpf/Makefile#L62).
+1. Install libbpf as a shared object in the system. Libbpf may already be
+   packaged for your distribution and, if not, you can build and install from
+   source. More info [here](https://github.com/libbpf/libbpf).
+1. Embed libbpf into your Go project as a vendored dependency. This means that
+   the libbpf code is statically linked into the resulting binary, and there
+   are no runtime dependencies.
+   [Tracee](https://github.com/aquasecurity/tracee) takes this approach.
+
+In the next sesssion you will find different ways to build libbpfgo.
+
+## Building
+
+Currently you will find the following GNU Makefile rules:
+
+| Makefile Rule &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; | Description &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; |
+|------------------------------|-----------------------------------|
+| **all**                      | builds libbpfgo (dynamic)         |
+| **install**                  | installs libbpfgo (dynamic)       |
```

Comment:
Can you add a comment about where it it installs libbpfgo to?

Reply:
To be honest, we don't have any binaries out of 'go make' process, install does not make much sense. I think the 'build' command already keeps objects cached for all the rest. If we do a 'go list' it shows 'stale':

```json
{
  "Dir": "/home/rafaeldtinoco/work/sources/ebpf/libbpfgo",
  "ImportPath": "github.com/aquasecurity/libbpfgo",
  "Name": "libbpfgo",
  "Root": "/home/rafaeldtinoco/work/sources/ebpf/libbpfgo",
  "Module": {
    "Path": "github.com/aquasecurity/libbpfgo",
    "Main": true,
    "Dir": "/home/rafaeldtinoco/work/sources/ebpf/libbpfgo",
    "GoMod": "/home/rafaeldtinoco/work/sources/ebpf/libbpfgo/go.mod",
    "GoVersion": "1.16"
  },
  "Match": [
    "."
  ],
  "Stale": true,
  "StaleReason": "not installed but available in build cache",
  "CgoFiles": [
    "libbpf_cb.go",
    "libbpfgo.go"
  ],
  "Imports": [
    "C",
    "fmt",
    "github.com/aquasecurity/libbpfgo/helpers",
    "net",
    "path/filepath",
    "strings",
    "sync",
    "syscall",
    "unsafe"
  ]
```

etc.. Should I remove the install and/or replace it by list ? I could have something as:

```diff
+libbpfgo-dynamic-list: libbpfgo-dynamic
+   CC=$(CLANG) \
+      CGO_CFLAGS=$(CGO_CFLAGS_DYN) \
+      CGO_LDFLAGS=$(CGO_LDFLAGS_DYN) \
+      go list -json . | $(JQ)
```

Reply:
Id say just remove it. If someone is interested in the output of go list they can run it themselves.